#!/usr/bin/env python
PACKAGE = "franka_example_controllers"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#          name                   paramtype level desciption                    default=? min  max
gen.add("NullSpaceImpedanceController", str_t, 0, "NullSpaceImpedanceController", "NullSpaceImpedanceController")

gen.add("task1_Kp_pos", double_t, 0, "task1_Kp_pos_gazebo", 100, 0, 600)#
gen.add("task1_Kp_ori", double_t, 0, "task1_Kp_ori_gazebo", 20,  0, 100)#

gen.add("task1_Kv_pos", double_t, 0, "task1_Kv_pos_gazebo", 0.1,  0, 100)#
gen.add("task1_Kv_ori", double_t, 0, "task1_Kv_ori_gazebo", 0.1,   0, 100)#

gen.add("task2_Md", double_t, 0, "inertia_gazebo", 3, 0, 100)#
gen.add("task2_Bd", double_t, 0, "damping_gazebo", 8, 0, 100)
gen.add("task2_Kd", double_t, 0, "stiffness_gazebo", 4, 0, 100)

gen.add("NullSpaceImpedanceEBObserverController", str_t, 0, "NullSpaceImpedanceEBObserverController", "NullSpaceImpedanceEBObserverController")

gen.add("P", double_t, 0, "P", 0.025, 0, 100)#
gen.add("K", double_t, 0, "K", 80,  0, 600)#

gen.add("Bv", double_t, 0, "Bv", 1, 0, 600)#
gen.add("Kd", double_t, 0, "Kd", 60,  0, 100)#

gen.add("ebPD_D", double_t, 0, "ebPD_D", 1, 0, 100)#
gen.add("ebPD_K", double_t, 0, "ebPD_K", 300,  0, 600)#

gen.add("Gamma_inv", double_t, 0, "Gamma_inv", 0, -100, 100)#

gen.add("NullSpaceImpedanceMBObserverController", str_t, 0, "NullSpaceImpedanceMBObserverController", "NullSpaceImpedanceMBObserverController")

gen.add("Kp", double_t, 0, "Kp", 200, 0, 600)#
gen.add("Kv", double_t, 0, "Kv", 1,  0, 600)#
gen.add("KI", double_t, 0, "KI", 8,  0, 600)#

gen.add("mbPD_D", double_t, 0, "mbPD_D", 10, 0, 100)#
gen.add("mbPD_K", double_t, 0, "mbPD_K", 1000,  0, 2000)#

#pkgname nodename name
exit(gen.generate(PACKAGE, "dynamic_compliance", "nullspace_impedance_controller_param"))
